tests/unit_tests/plugin_registries/test_plugin_registries.py

```diff
@@ -41,25 +231,25 @@ def generate_id(fpath):
 
     def test_registered_plugin_property(self):
         """Ensure registered_plugins is valid in its nature."""
-        print(self.plugin_registry.registered_plugins)
-        assert isinstance(self.plugin_registry.registered_plugins, dict)
-        assert "yaml_based" in self.plugin_registry.registered_plugins
-        assert "module_based" in self.plugin_registry.registered_plugins
-        assert "text_based" in self.plugin_registry.registered_plugins
+        print(self.pr_validator.registered_plugins)
+        assert isinstance(self.pr_validator.registered_plugins, dict)
+        assert "yaml_based" in self.pr_validator.registered_plugins
+        assert "module_based" in self.pr_validator.registered_plugins
+        assert "text_based" in self.pr_validator.registered_plugins
```

Comment:
This is just updated so now it uses the internal PR Validator class rather than the imported plugin_registry validator form geoips/plugin_registry.py, right?

Reply:
Correct. The the PluginRegistry class doesn't really do any validation anymore. It has all been moved here.

Now that I think about it, though, we should test what happens now if the plugin registry is corrupted or malformed. Are the resulting errors helpful? If not, can we make them helpful?

@evrose54 What do you think on this?

Reply:
That's probably a good idea. I don't love that the Plugin Registry isn't validated during it's creation, but it will check to make sure that every plugin added is not duplicately named and that every plugin added is valid in its nature. The way I see it we have two options:

1. Run the validation routines when the plugin registry is created, and remove references to pytest
2. Keep the validation routines as a unit test, however run some final validation at the end of the registry's creation to ensure it's in a correct state. 

A side note:
When we load in the registry to use in GeoIPS, via get_plugin, ... it might be smart to perform some validation of the contents it's reading. Users could possible modify the registry files manually and we should check that the registries are in fact valid when loading them in. Currently we're assuming that after they've been created they are valid, and do no validation during runtime.

Reply:
I'm debating whether I think this is safe. I'm not sure if I'm comfortable only validating at creation time, but maybe it is okay.

Maybe another option. Would it be possible to separate the validation routines from the pytest calls? Keep some or all of the validation in the original class, but have it raise errors rather than call pytest.xfail, then catch those errors in the unit tests?

I think that I'm good with approving this PR, but I think we should create another issue to think about where validation should actually occur and how to handle unit testing for this. For example, maybe the part that should be getting tested in the unit tests is actually the validation functions themselves.

Reply:
That's a fair point. I'll create an issue that we should discuss where the validation should occur, and what the unit tests should actually test. Once that is done, I'll merge this PR.

Reply:
Going to merge now, see issue #496 for more information.